### PR TITLE
Sprint 4

### PR DESCRIPTION
**Pull request by AlinaNikolaienko**
Tests for user story "Koel App | All songs", Internship-82790

### CODE REVIEW

File: src/test/java/tests/AllSongsTests.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+        AllSongsPage allSongsPage = new AllSongsPage(driver);
+
+        loginPage.login();
+        //Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
+
+        homePage.clickAllSongs();
+        //String songsURL = "https://qa.koel.app/#!/songs";
+        //Assert.assertEquals(driver.getCurrentUrl(), songsURL);
```
Comment (Berichicko) [inline src/test/java/tests/AllSongsTests.java]: I recommend not using code comments: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.

File: src/test/java/tests/AllSongsTests.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+        AllSongsPage allSongsPage = new AllSongsPage(driver);
+
+        loginPage.login();
+        //Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
+
+        homePage.clickAllSongs();
+        String songsURL = "https://qa.koel.app/#!/songs";
+        Assert.assertEquals(driver.getCurrentUrl(), songsURL);
```
Comment (Berichicko) [inline src/test/java/tests/AllSongsTests.java]: **One test - One Assert**

For example

Clean_Test

File: src/test/java/pages/SearchPage.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        return nameArtist.getText();
+    }
+
+    public boolean getNameAlbum(String songName) {
+        findElement(nameAlbum);
+        return nameAlbum.getText().contains(songName);
+    }
+*/
+
+}
```
Comment (Berichicko) [inline src/test/java/pages/SearchPage.java:90]: I recommend not using code comments: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.

File: src/test/java/tests/AlbumsTests.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+public class AlbumsTests extends BaseTest {
+    @Test
+    public void countSongsInPlaylist() {
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+        AllSongsPage allSongsPage = new AllSongsPage(driver);
+
+        loginPage.login();
+        //Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
```
Comment (Berichicko) [inline src/test/java/tests/AlbumsTests.java]: **Use in test methods, the following logical split:   GIVEN – WHEN - THEN**
    
For example

```
 @Test

    public void testChangeProfileName (){
        // GIVEN

        String newName  = getRandomString();

        String currentPassword = "te$t$tudent";

        provideEmail("demo@class.com");

        providePassword();

        clickSubmitBtn();

 

        // WHEN

        openUserProfilePage();

        setName(newName);

        setCurrentPassword(currentPassword);

 

 

        // THEN

        Assert.assertTrue(getSuccessPopUp().isDisplayed());

}
```

Please, add Assert

**One test - One Assert**

**Comment by Berichicko** [comment]
You got this problem, please correct it.

image

image

image

HINT

How to analyze the results of tests:

Test status: 
Test status_en

	
Test error (red). This status is assigned to tests that caused an exception from the tested source code.
Test error" means there's a problem with the code being tested. This is not okay for a tester. We need to fix it to make sure the software works correctly
Test error_en

Test failed(yellow). If at least one test receives this status, then all its parents are marked as failed.
Test failed_en

**Comment by AlinaNikolaienko** [comment]
Hi Alex @Berichicko, thank you for your recommendations.
Unfortunately or luckily my tests are failing due to Assertion, not because of NoSuchElementError. I switch css to xpath locator. Hope it will help with errors.
I resubmitted my last update, please check it.

**Comment by Berichicko** [comment]
> Hi Alex @Berichicko, thank you for your recommendations. Unfortunately or luckily my tests are failing due to Assertion, not because of NoSuchElementError. I switch css to xpath locator. Hope it will help with errors. I resubmitted my last update, please check it.

Hi dear Alina

I run tests and result

image

image

**Pull request closed without merging**